frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx

```diff
@@ -79,27 +80,45 @@ export const ActionParameterMappingForm = ({
 
   return (
     <div>
-      {actionParameters.map((actionParam: WritebackParameter) => (
-        <ParameterFormSection key={actionParam.id}>
-          <ParameterFormLabel>
-            {actionParam.name ?? actionParam.id}
-          </ParameterFormLabel>
-          <Select
-            value={currentMappings[getTargetKey(actionParam)] ?? null}
-            onChange={(e: SelectChangeEvent<string>) =>
-              handleParameterChange(e?.target?.value, actionParam.target)
-            }
-            options={[
-              { name: t`Ask the user`, value: null },
-              ...dashboardParameters.map(dashboardParam => ({
-                key: dashboardParam.id,
-                name: dashboardParam.name,
-                value: dashboardParam.id,
-              })),
-            ]}
-          />
-        </ParameterFormSection>
-      ))}
+      {actionParameters.map((actionParam: WritebackParameter) => {
+        const isHidden =
+          !!action?.visualization_settings?.fields?.[actionParam.id]?.hidden;
+        const mappedValue = currentMappings[getTargetKey(actionParam)];
+        const showError = !mappedValue && isHidden && !!actionParam.required;
+
+        return (
+          <ParameterFormSection
+            key={actionParam.id}
+            data-testid={`parameter-form-section-${actionParam.id}`}
+          >
+            <ParameterFormLabel error={showError}>
+              {actionParam.name ?? actionParam.id}{" "}
+              {showError && <span>{t`is required`}</span>}{" "}
+              {isHidden && <ParameterFormBadge>{t`Hidden`}</ParameterFormBadge>}
```

Comment:
The first thing I noticed is that the badge is unusually close to the text.
I don't think that `{" "}` space is enough of a visual separator and I think we're going to need CSS here.

In Figma there is 8px of space between the badge and text and it looks much more natural:

![image](https://github.com/metabase/metabase/assets/6830683/4a167224-f7c7-429c-9c0c-fc7a63d05407)

vs here:

![image](https://github.com/metabase/metabase/assets/6830683/076f824f-e3bc-45f4-b5ce-8e3faf805cba)

Reply:
nice catch, I'll fix it!

Reply:
should be fixed now. I think the space is too large btw